### PR TITLE
zimenkov code

### PR DESCRIPTION
**Pull request by tetroboy**
refactor and check my code

### CODE REVIEW

File: dima kod/test.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+#include <vector>
+#include <utility>
+#include <algorithm>
+
```
Comment (V1zav) [inline dima kod/test.cpp:5]: Function to find the minimum number of announcement times required to notify all classes, given their start and end times

File: dima kod/test.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+std::vector<int> find_pa_times(const std::vector<std::pair<int, int>>& classes) {
+    std::vector<int> pa_times;
+
```
Comment (V1zav) [inline dima kod/test.cpp:10]: If no classes are provided, return an empty vector

File: dima kod/test.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+std::vector<int> find_pa_times(const std::vector<std::pair<int, int>>& classes) {
+    std::vector<int> pa_times;
```
Comment (V1zav) [inline dima kod/test.cpp]: std::vector<int> announcement_times;

File: dima kod/test.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    if (classes.empty()) {
+        return pa_times;        // returning clear vector if amount of classes is 0
```
Comment (V1zav) [inline dima kod/test.cpp]: announcement_times

File: dima kod/test.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    std::vector<std::pair<int, int>> sorted_classes = classes;
+    std::sort(sorted_classes.begin(), sorted_classes.end(), [](const auto& a, const auto& b) {
+        return a.second < b.second;
+    });
+
+    // Starting from the end of first class
+    int last_end_time = sorted_classes[0].second;
+
+    // Pushing to our vector 1 announcement time
+    pa_times.push_back(last_end_time);
```
Comment (V1zav) [inline dima kod/test.cpp]: announcement_times

File: dima kod/test.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (V1zav) [inline dima kod/test.cpp]: Iterate through the sorted classes to find all necessary announcement times

File: dima kod/test.cpp [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    // Checking if there are classes which are still not notified do the same logic as previous
+    for (size_t i = 1; i < sorted_classes.size(); ++i) {
+        int start_time = sorted_classes[i].first;
+        if (start_time >= last_end_time) {
+            last_end_time = sorted_classes[i].second;
+            pa_times.push_back(last_end_time);
```
Comment (V1zav) [inline dima kod/test.cpp]: announcement_times

File: dima kod/test.cpp [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    }
+
+    return pa_times;
```
Comment (V1zav) [inline dima kod/test.cpp]: announcement_times

File: dima kod/test.cpp [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+int main() {
+    // test condition
+    std::vector<std::pair<int, int>> classes = { {1, 20},
+        {14, 17}, {12, 18}, {22, 30}, {27, 42},
+        {20, 36}, {33, 56}, {40, 60} };
+
+    std::vector<int> pa_times = find_pa_times(classes);
```
Comment (V1zav) [inline dima kod/test.cpp]: std::vector<int> announcement_times = find_minimum_announcement_times(classes);

File: dima kod/test.cpp [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+    std::cout << "min amount of announcements: " << pa_times.size() << " at time: ";
```
Comment (V1zav) [inline dima kod/test.cpp]: std::cout << "Minimum number of announcements: " << announcement_times.size() << " at times: ";

File: dima kod/test.cpp [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    for (int time : pa_times) {
```
Comment (V1zav) [inline dima kod/test.cpp]: int time : announcement_times

File: dima kod/test.cpp (hunk 1 above)
Comment (tetroboy) [reply dima kod/test.cpp:5]: ok

File: dima kod/test.cpp (hunk 2 above)
Comment (tetroboy) [reply dima kod/test.cpp:10]: ok

File: dima kod/test.cpp (hunk 3 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 4 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 5 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 6 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 7 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 8 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 9 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 10 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

File: dima kod/test.cpp (hunk 11 above)
Comment (tetroboy) [reply dima kod/test.cpp]: ok

**Pull request merged**